### PR TITLE
Online Book Store

### PR DESCRIPTION
**Pull request by Oleksandr-Tymoshenko**

### CODE REVIEW

File: src/main/java/book/store/onlinebookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "book")
```
Comment (ystankevych) [inline src/main/java/book/store/onlinebookstore/model/Book.java]:
```suggestion
@Table(name = "books")
```
table for books

File: src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        sessionFactory.inTransaction(session -> session.persist(book));
+        return book;
```
Comment (ystankevych) [inline src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
try{
        sessionFactory.inTransaction(session -> session.persist(book));
        return book;
        } catch (Exception e) {
        throw new yourSomeException(message, e);
        }

```

File: src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return sessionFactory.openSession().createQuery("from Book ", Book.class).getResultList();
```
Comment (ystankevych) [inline src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
        return sessionFactory.openSession().createQuery("from Book ", Book.class).getResultList();

```
why whithout try/catch?) Even if the chance of an exception is very small, these methods can throw an exception.

**Review by ystankevych — COMMENTED** [review]
Hi! Great job! I approved, but left some comments (Pay attention to them =) )

**Review by ystankevych — APPROVED** [review]

File: src/main/java/book/store/onlinebookstore/model/Book.java (hunk 1 above)
Comment (Elena-Bruyako) [reply src/main/java/book/store/onlinebookstore/model/Book.java]: Agree

File: src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java]: agree

File: src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java (hunk 3 above)
Comment (Elena-Bruyako) [reply src/main/java/book/store/onlinebookstore/repository/impl/BookRepositoryImpl.java]: agree

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Review by Andriy-dz — APPROVED** [review]

**Pull request merged**